sdk/src/client/api/block_builder/input_selection/requirement/account.rs

```diff
@@ -3,39 +3,17 @@
 
 use super::{Error, InputSelection, Requirement};
 use crate::{
-    client::{api::input_selection::Burn, secret::types::InputSigningData},
-    types::block::output::{AccountId, AccountTransition, Output, OutputId},
+    client::secret::types::InputSigningData,
+    types::block::output::{AccountId, Output, OutputId},
 };
 
-pub(crate) fn is_account_transition<'a>(
-    input: &Output,
-    input_id: OutputId,
-    outputs: &[Output],
-    burn: impl Into<Option<&'a Burn>>,
-) -> Option<AccountTransition> {
-    if let Output::Account(account_input) = &input {
-        let account_id = account_input.account_id_non_null(&input_id);
-        // Checks if the account exists in the outputs and gets the transition type.
-        for output in outputs.iter() {
-            if let Output::Account(account_output) = output {
-                if *account_output.account_id() == account_id {
-                    if account_output.state_index() == account_input.state_index() {
-                        // Governance transition.
-                        return Some(AccountTransition::Governance);
-                    } else {
-                        // State transition.
-                        return Some(AccountTransition::State);
-                    }
-                }
-            }
-        }
-        if let Some(burn) = burn.into() {
-            if burn.accounts().contains(&account_id) {
-                return Some(AccountTransition::Governance);
-            }
-        }
+/// Checks if an output is an account with output ID that matches the given account ID.
+pub(crate) fn is_account_with_id(output: &Output, account_id: &AccountId, output_id: &OutputId) -> bool {
```

Comment:
what happens if `OutputId` doesn't belong to `Output`?

Reply:
Not much, output is just used to check the type. But that can't happen anyway.